nntrainer/tensor/tensor.cpp

```diff
@@ -795,6 +796,61 @@ std::vector<Tensor> Tensor::split(unsigned num_size, int axis) {
   return ret;
 }
 
+Tensor Tensor::cat(const std::vector<Tensor> &tensors, int axis) {
+  if (axis == -1) {
+    axis = 3;
+  }
+  NNTR_THROW_IF(tensors.empty(), std::invalid_argument)
+    << "given tensor vector is empty";
+  auto ref_dim = tensors.front().getDim();
+  ref_dim.setTensorDim(axis, 1);
+  NNTR_THROW_IF(!std::all_of(tensors.begin(), tensors.end(),
+                             [&ref_dim, axis](const Tensor &t) {
+                               auto cur_dim = t.getDim();
+                               cur_dim.setTensorDim(axis, 1);
+                               return ref_dim == cur_dim;
+                             }),
+                std::invalid_argument)
+    << " all tensor must have the same dimension except for the axis, ref_dim: "
+    << ref_dim << " axis : " << axis;
+
+  auto axis_dim = 0u;
+  for (auto &t : tensors) {
+    axis_dim += t.getDim().getTensorDim(axis);
+  }
+
+  auto iter_value = [](std::array<unsigned, 4> &loc,
+                       std::array<unsigned, 4> &start_loc, Tensor &t,
+                       const TensorDim &ref_dim) -> float & {
+    auto &value = t.getValue(loc[0], loc[1], loc[2], loc[3]);
+    for (int i = 3; i >= 0; --i) {
+      loc[i]++;
+      if (loc[i] - start_loc[i] == ref_dim.getTensorDim(i)) {
+        loc[i] = start_loc[i];
+        continue;
+      }
+      break;
+    }
+    return value;
+  };
+
+  auto ret_dim = ref_dim;
+  ret_dim.setTensorDim(axis, axis_dim);
+
+  auto ret = Tensor(ret_dim);
+
+  std::array<unsigned, 4> loc = {0, 0, 0, 0};
+  for (auto &t : tensors) {
+    std::array<unsigned, 4> start_loc = loc;
+    for (auto i = 0u, sz = t.size(); i < sz; ++i) {
+      iter_value(loc, start_loc, ret, t.getDim()) = t.getValue(i);
```

Comment:
I would highly recommend you to refer the concat layer implementation for higher efficiency.

Reply:
I'd like to forward discussion to https://github.com/nnstreamer/nntrainer/pull/1792/commits/7d0d1624df8b11244955a6ec12564eb67eb94ba8#r770239923.

Concat layer implementation can't be directly used here as this implementation support different axis as well, 
eg) Tensor::cat( [3, 2, 4 ,5] , [3, 4, 4, 5], axis = 1 )  results is [3, 6, 4, 5]

Reply:
Resolved offline that this is supported. I'll check when I improve those :)